src/notifications/generate.ts

```diff
@@ -127,12 +128,27 @@ export const notificationTitleMap: Record<
 
     return `New post from <b>${userName}</b>, check it out now!`;
   },
+  user_given_top_reader: (ctx: NotificationUserTopReaderContext) => {
+    const keyword = ctx.keyword.flags.title;
+    return `Great news! You earned the top reader badge in ${keyword}.`;
+  },
 };
 
 export const generateNotificationMap: Record<
   NotificationType,
   (builder: NotificationBuilder, ctx: never) => NotificationBuilder
 > = {
+  user_given_top_reader: (builder, ctx: NotificationUserTopReaderContext) =>
+    builder
+      .icon(NotificationIcon.TopReaderBadge)
+      .referenceUserTopReader(ctx.userTopReader)
+      .targetUrl(notificationsLink)
+      .setTargetUrlParameter([
```

Comment:
Oh I see you do it here, maybe we should do this in the builder rather?

Reply:
Not sure what you mean? Doing it like this for `streak_reset_restore`, `promoted_to_admin` and `promoted_to_moderator` notifications.

Reply:
Oh right, I thought we set it in the builder already, that case never mind me :)